15: Rich Text Editor Tiptap

sofietoft opened the pull request:
## Description

Umbraco 15 introduces a new Rich Text Editor UI, Tiptap.

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

15.0.0

## Deadline (if relevant)

November 14th, or anytime before.

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/configuration.md:
```diff
@@ -0,0 +1,69 @@
+
+## Dimensions
+
+Define `height` and `width` of the editor displayed in the content section.
```
```suggestion
Define the `height` and `width` of the editor displayed in the Content section.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/configuration.md:
```diff
@@ -0,0 +1,69 @@
+
+### Inline
+
+In this mode the toolbar is hidden, and only shows up when content in the editor is highlighted.
```
```suggestion
In this mode, the toolbar is hidden and only shows up when content in the editor is highlighted.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/configuration.md:
```diff
@@ -0,0 +1,69 @@
+
+## Blocks
+
+Blocks can be added as elements in the Rich Text Editor. Configuration and rendering of Blocks is described in the [Blocks in Rich Text Editor](blocks.md) article.
```
```suggestion
Blocks can be added as elements in the Rich Text Editor. Configuration and rendering of Blocks are described in the [Blocks in Rich Text Editor](blocks.md) article.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/configuration.md:
```diff
@@ -0,0 +1,69 @@
+
+## Ignore User Start Nodes
+
+Some of the backoffice users might be restricted to a specific part of the content tree. When the "Ignore User Start Nodes" is checked, users can pick any piece of content, when adding internal links.
```
```suggestion
Some backoffice users might be restricted to specific parts of the Content tree. When the "Ignore User Start Nodes" option is checked, users can pick any piece of content, when adding internal links.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/configuration.md:
```diff
@@ -0,0 +1,69 @@
+
+## Image Upload Folder
+
+Images added through the RTE is by default added to the root of the Media library.
```
```suggestion
Images added through the RTE are by default added to the root of the Media library.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/README.md:
```diff
@@ -1,69 +1,61 @@
-
-{% hint style="info" %}
-**Are you using custom configuration or plugins with TinyMCE?**
+In Umbraco 15, the Rich Text Editor has gotten an alternative that introduces Tiptap as the new default property editor UI.
```
```suggestion
In Umbraco 15, the Rich Text Editor has a new default property editor UI that introduces Tiptap as an alternative.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/README.md:
```diff
@@ -1,69 +1,61 @@
+The Tiptap UI currently does not support using custom styles for your rich text.
 {% endhint %}
 
+The Rich Text Editor (RTE) Tiptap property editor is based on [Tiptap](https://tiptap.dev/) and is highly configurable. Dependening on the configuration set up, it gives your editors a lot of flexibility when working with content.
```
```suggestion
The Rich Text Editor (RTE) Tiptap property editor is highly configurable and based on [Tiptap](https://tiptap.dev/). Depending on the configuration setup, it provides editors a lot of flexibility when working with content.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/README.md:
```diff
@@ -1,69 +1,61 @@
-## [Blocks](rte-blocks.md)
+## [Blocks](blocks.md)
 
 Use Blocks to define specific parts which can be added as part of the markup of the Rich Text Editor.
```
```suggestion
Use Blocks to define specific parts that can be added as part of the markup of the Rich Text Editor.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/blocks.md:
```diff
@@ -0,0 +1,104 @@
+The Data Type editor allows you to configure the following properties:
+
+* **Available Blocks** - Here you will define the Block Types to be available for use in the property. Read more on how to set up Block Types below.
+* **Blocks Live editing mode** - Enabling this will make editing of a Block happen directly to the Rich Text Editor, making changes appear as you type.
```
```suggestion
* **Blocks Live editing mode** - Enabling this will edit a Block directly to the Rich Text Editor, making changes appear as you type.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/blocks.md:
```diff
@@ -0,0 +1,104 @@
+
+If you use ModelsBuilder, you can specify the Content Model (and optionally the Settings Model) in the Partial View model. This allows for type-safe access to the Block data.
+
+The following example shows a Partial View of a Block with `MyBlockType` as Content Model and `MyBlockSettingsType` as Settings Model:
```
```suggestion
The following example shows a Partial View of a Block with `MyBlockType` as the Content Model and `MyBlockSettingsType` as the Settings Model:
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/blocks.md:
```diff
@@ -0,0 +1,104 @@
+
+## Build a Custom Backoffice View
+
+Building Custom Views for Block representations in Backoffice is the same for all Block Editors. [Read about building a Custom View for Blocks here](../../../../../tutorials/creating-custom-views-for-blocklist.md)
```
```suggestion
Building Custom Views for Block representations in Backoffice is the same for all Block Editors. [Read about building a Custom View for Blocks here](../../../../../tutorials/creating-custom-views-for-blocklist.md).
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/change-rich-text-editor-ui.md:
```diff
@@ -0,0 +1,57 @@
+# Change Rich Text Editor UI
+
+{% hint style="info" %}
+Umbraco 15 ships with two UI options for working with rich text, Tiptap and TinyMCE. The latter will eventually be removed from the CMS.
```
```suggestion
Umbraco 15 includes two options for rich text editing: Tiptap and TinyMCE. TinyMCE will eventually be phased out of the CMS.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/change-rich-text-editor-ui.md:
```diff
@@ -0,0 +1,57 @@
+{% hint style="info" %}
+Umbraco 15 ships with two UI options for working with rich text, Tiptap and TinyMCE. The latter will eventually be removed from the CMS.
+
+This article will guide you through switching from using the TinyMCE UI to using the new Tiptap UI.
```
```suggestion
This article will guide you through switching from TinyMCE UI to the new Tiptap UI.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/change-rich-text-editor-ui.md:
```diff
@@ -0,0 +1,57 @@
+
+## Verify your content
+
+When you have updated all the relevant Document Types, it is recommend to verify the content that uses the rich text editor.
```
```suggestion
When you have updated all the relevant Document Types, it is recommended to verify the content that uses the Rich Text Editor.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/change-rich-text-editor-ui.md:
```diff
@@ -0,0 +1,57 @@
+
+The first step in this guide is to create a new Rich Text Editor Data Type using the Tiptap UI.
+
+As an alternative, you can create a new Data Type with the Tiptap UI when updating the Document Type.
```
Umm, I'm not sure if Tiptap UI is the correct term cz we are not seeing any user interface as such other than selecting a data type/property editor.

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration.md:
```diff
@@ -1,71 +1,46 @@
+![Toolbar: All options enabled](images/rte-tiptap-all-toolbar-items.png)
 
-In the examble above, all 34 options have been enabled. The options include copy/paste buttons, font styles like bold and italics, bullet lists and options to embed videos and insert images.
+In the examble above, all 27 options have been enabled. The options include font styles like bold and italics, bullet lists and options to embed videos and insert images.
```
```suggestion
In the example above, all 27 options have been enabled. These options include font styles like bold and italics, bullet lists, and options to embed videos and insert images.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration.md:
```diff
@@ -1,71 +1,46 @@
+![Enhance and customize the capabilities of the Rich Text Editor toolbar](images/rte-tiptap-capabilities-and-toolbar.png)
 
-The RTE styles are defined in CSS files which can be created in the **Settings** section. Read the [RTE Styles](rte-styles.md) article to learn more about this feature.
+* Enhance the capabilities of the toolbar by enablding or disabling extentions.
```
```suggestion
* Enhance the capabilities of the toolbar by enabling or disabling extensions.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration.md:
```diff
@@ -1,71 +1,46 @@
+## Available Blocks
 
-Blocks can be added as elements in the Rich Text Editor. Configuration and rendering of Blocks is described in the [Blocks in Rich Text Editor](rte-blocks.md) article.
+Blocks can be added as elements in the Rich Text Editor. Configuration and rendering of Blocks is described in the [Blocks in Rich Text Editor](blocks.md) article.
```
```suggestion
Blocks can be added as elements in the Rich Text Editor. Configuration and rendering of Blocks are described in the [Blocks in Rich Text Editor](blocks.md) article.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration.md:
```diff
@@ -1,71 +1,46 @@
-Select the width of the link picker overlay. The overlay size comes in three sizes: Small, Medium, and Large.
-
-### Hide Label
+Images added through the RTE is by default added to the root of the Media library.
```
```suggestion
Images added through the RTE are by default added to the root of the Media library.
```

eshanrnh commented on 15/umbraco-cms/fundamentals/data/data-types/default-data-types.md:
```diff
@@ -99,25 +99,9 @@ This Data type enables editors to choose from a list of radiobuttons.
 ## Richtext Editor
 
-The TinyMCE based WYSIWYG editor. This is the standard editor used to edit larger amount of text. The editor has a lot of settings, which can be changed on the Richtext editor Data Type in the Settings section.
+A TipTap based What You See Is What You Get (WYSIWYG) editor. This is the standard editor used to edit larger amount of text. The editor has a lot of settings, which can be changed on the Richtext editor Data Type in the Settings section.
```
```suggestion
A TipTap-based What You See Is What You Get (WYSIWYG) editor. This is the standard editor used to edit a larger amount of text. The editor has a lot of settings, which can be changed on the Richtext editor Data Type in the Settings section.
```

eshanrnh reviewed (approved):
Looks good to me, @sofietoft 🙌 Just some minor suggestions w.r.t grammar, spellings. Nothing major 😀

sofietoft commented:
Thanks for the review @eshanrnh !

I've made some changes based on your suggestions - all images and links should be correct now.
And I've changed the wording a bit in the article about swapping the property editors.

Let me know what you think!

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/styles.md:
```diff
@@ -1,4 +1,4 @@
+# Styles
 
 It is possible to define specific styles and fonts for the Rich Text Editor (RTE). Once you have defined the styles, and enabled them on the RTE Data Type, the styles can be accessed directly in the Content section.
 
```
This image is not rendering in the preview.

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor-tinymce/plugins.md:
```diff
@@ -2,19 +2,17 @@
+## Premium Plugins
 
 TinyMCE also has a number of [premium plugins](https://www.tiny.cloud/docs/tinymce/6/plugins/#premium-plugins) available. These plugins are only available for [paid TinyMCE subscriptions](https://www.tiny.cloud/pricing/). They can be added to the rich text editor by [adding a bit of configuration](#adding-a-premium-plugin).
 
```
I know you didn't change the text here but noticed these lists need space after the number 😀 
1.Add a manifest file
2.Add the plugin.js file

eshanrnh commented on 15/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/rich-text-editor/configuration.md:
```diff
@@ -1,71 +1,46 @@
+![Enhance and customize the capabilities of the Rich Text Editor toolbar](images/rte-tiptap-capabilities-and-toolbar.png)
 
-The RTE styles are defined in CSS files which can be created in the **Settings** section. Read the [RTE Styles](rte-styles.md) article to learn more about this feature.
+* Enhance the capabilities of the toolbar by enabling or disabling extensions.
```
This list seems out of place cz it's right below an image without a leading line.

eshanrnh reviewed (commented):
Looks good to me, @sofietoft 🙌 Just noticed some more things while previewing the PR.

sofietoft commented:
Thanks again @eshanrnh  💪 

Now I've finally caught everything.
In the configuration article I moved the list above the image, and updated the text before it slightly.
Let me know what you think! 🤞

eshanrnh commented:
Looks perfect 🙌 Thanks for applying the suggestions 💪
